patches/minecraft/net/minecraft/block/Block.java.patch

```diff
@@ -65,7 +65,27 @@
        }
     }
  
-@@ -553,7 +554,7 @@
+@@ -493,14 +494,11 @@
+    @Deprecated
+    public List<ItemStack> func_220076_a(BlockState p_220076_1_, LootContext.Builder p_220076_2_) {
+       ResourceLocation resourcelocation = this.func_220068_i();
+-      if (resourcelocation == LootTables.field_186419_a) {
+-         return Collections.emptyList();
+-      } else {
+-         LootContext lootcontext = p_220076_2_.func_216015_a(LootParameters.field_216287_g, p_220076_1_).func_216022_a(LootParameterSets.field_216267_h);
+-         ServerWorld serverworld = lootcontext.func_202879_g();
+-         LootTable loottable = serverworld.func_73046_m().func_200249_aQ().func_186521_a(resourcelocation);
+-         return loottable.func_216113_a(lootcontext);
+-      }
++      p_220076_2_ = p_220076_2_.func_216015_a(LootParameters.field_216287_g, p_220076_1_);
++      ServerWorld serverworld = p_220076_2_.func_216018_a();
++      LootTable loottable = serverworld.func_73046_m().func_200249_aQ().func_186521_a(resourcelocation);
++
++      return net.minecraftforge.event.ForgeEventFactory.fireBlockLootEvents(loottable, p_220076_2_);
```

Comment:
Hmm, so this currently fires both events, but if someone overrides this method and does something like "call super to generate table drops, then add some dynamic stuff", this won't capture that second step.

I would leave the Generate event like it is now, but move the Drop event to the places that call this method, so it has full access to *all* drops, from loot table or otherwise.

Reply:
Removed with the newest commit, I added back in the drop capture behaviour with some small changes as that seemed the most reasonable way to do it. This does increase the patch sizes by a decent amount bit which kind of sucks but it also makes sure every block drop case is well covered.